File: bms/go/geo/geo_s1_Interval_InteriorContains/geo_s1_Interval_InteriorContains.go
package test

import (
	"math"
)

func (i geo_s1_Interval) InteriorContains(p float64) bool {
	if p == -math.Pi {
		p = math.Pi
	}
	if i.IsInverted() {
		return p > i.Lo || p < i.Hi
	}
	return (p > i.Lo && p < i.Hi) || i.IsFull()
}

func (i geo_s1_Interval) IsInverted() bool { return i.Lo > i.Hi }

func (i geo_s1_Interval) IsFull() bool { return i.Lo == -math.Pi && i.Hi == math.Pi }

type geo_s1_Interval struct {
	Lo, Hi float64
}

File: translations_only/fail/go/geo/geo_s1_Interval_InteriorContains/mistral/CAPR/geo_s1_Interval_InteriorContains.rs


use std::f64;

struct GeoS1Interval {
    lo: f64,
    hi: f64,
}

fn interior_contains(i: &GeoS1Interval, p: &mut f64) -> bool {
    if *p == -f64::consts::PI {
        *p = f64::consts::PI;
    }
    if i.is_inverted() {
        return *p > i.lo || *p < i.hi;
    }
    (*p > i.lo && *p < i.hi) || i.is_full()
}

impl GeoS1Interval {
    fn is_inverted(&self) -> bool {
        self.lo > self.hi
    }

    fn is_full(&self) -> bool {
        self.lo == -f64::consts::PI && self.hi == f64::consts::PI
    }
}

